R/best_and_worst_cases.R

```diff
@@ -0,0 +1,23 @@
+best_and_worst_cases <- function(data) {
+  data |>
+    mutate(amount_of_distinct_products = dplyr::n_distinct(.data$ep_product), .by = "companies_id") |>
+    mutate(equal_weight = (1/.data$amount_of_distinct_products)) |>
+    lowest_risk_category_per_company_benchmark(.by = c("companies_id", "benchmark")) |>
+    highest_risk_category_per_company_benchmark(.by = c("companies_id", "benchmark")) |>
+    mutate(dummy_best = ifelse(.data$emission_profile == .data$min_risk_category_per_company_benchmark, 1, 0)) |>
```

Comment:
This function depends on crucial columns from the data, such `ep_product`, "companies_id", "benchmark", etc. Remember to test for graceful errors when those crucial columns don't exist.